explainer/index.bs

```diff
@@ -793,6 +793,43 @@ Some alternatives are mentioned in issue [#747](https://github.com/gpuweb/gpuweb
 
 ## Image, Video, and Canvas input ## {#image-input}
 
+Issue: Exact API still in flux as of this writing.
+
+WebGPU is largely isolated from the rest of the Web platform, but has several interop points.
+One of these is image data input into the API.
+Aside from the general data read/write mechanisms (`writeTexture`, `writeBuffer`, and `mapAsync`),
+data can also come from `<img>`/`ImageBitmap`, canvases, and videos.
+There are many use-cases that require these, including:
+
+- Initializing textures from encoded images (JPEG, PNG, etc.)
+- Rendering text with 2D canvas for use in WebGPU.
+- Video element and video camera input for image processing, ML, 3D scenes, etc.
+
+There are two paths:
+
+- `copyExternalImageToTexture()` copies color data from a sub-rectangle of an
+    image/video/canvas object into an equally-sized sub-rectangle of a `GPUTexture`.
+    The input data is captured at the moment of the call.
+- `importTexture()` takes a video or canvas and creates a `GPUExternalTexture` object which *can*
+    provide direct read access to an underlying resource if it exists on the (same) GPU already,
+    avoiding unnecessary copies or CPU-GPU bandwidth.
+    This is typically true of hardware-decoded videos and most canvas elements.
+
+Issue: Update both names to whatever we settle on.
+
+### GPUExternalTexture ### {#image-input-external-texture}
+
+A `GPUExternalTexture` is a read-only texture object which can be used in similar ways to normal
```

Comment:
the "read-only" and "immutable" characterizing this texture read a little bit weird to me, since it's a "sampled" texture, and all sampled textures are like this. Moreover, "read-only" is confusing with "readonly-storage", which this texture isn't.